src/main/java/aget/periodsbot/bot/PeriodsBot.java

```diff
@@ -1,59 +1,129 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.telegram.telegrambots.bots.TelegramLongPollingBot;
 import org.telegram.telegrambots.meta.TelegramBotsApi;
 import org.telegram.telegrambots.meta.api.objects.Update;
+import org.telegram.telegrambots.meta.bots.AbsSender;
 import org.telegram.telegrambots.meta.exceptions.TelegramApiException;
 import org.telegram.telegrambots.updatesreceivers.DefaultBotSession;
 
-public class PeriodsBot extends TelegramLongPollingBot {
-    private static final Logger log = LoggerFactory.getLogger(PeriodsBot.class);
+/**
+ * Main bot class.
+ *
+ * @since 0.1.0
+ */
+public final class PeriodsBot extends TelegramLongPollingBot {
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(PeriodsBot.class);
+
+    /**
+     * Bot properties.
+     */
     private final BotProps props;
-    private final TelegramBotsApi telegramBotsApi;
-    private final Route<Update, Send> route;
 
-    public PeriodsBot(BotProps props,
-                      TelegramBotsApi telegramBotsApi,
-                      Route<Update, Send> route) {
-        this.props = props;
-        this.telegramBotsApi = telegramBotsApi;
-        this.route = route;
+    /**
+     * Telegram bots API.
+     */
+    private final TelegramBotsApi api;
+
+    /**
+     * Route.
+     */
+    private final Route<Update, AbsSender> route;
+
+    public PeriodsBot(
+        final BotProps props,
+        final Route<Update, AbsSender> route
+    ) throws TelegramApiException {
+        this(props, new TelegramBotsApi(DefaultBotSession.class), route);
     }
 
-    public PeriodsBot(BotProps props, Route<Update, Send> route) throws TelegramApiException {
+    public PeriodsBot(
+        final BotProps props,
+        final TelegramBotsApi api,
+        final Route<Update, AbsSender> route
+    ) {
+        super(props.botToken());
         this.props = props;
-        this.telegramBotsApi = new TelegramBotsApi(DefaultBotSession.class);
+        this.api = api;
         this.route = route;
     }
 
     @Override
-    public void onUpdateReceived(Update update) {
-        this.route.route(update).ifPresent(
-                send -> send.send(this)
-        );
+    public void onUpdateReceived(final Update update) {
+        this.route.route(new TgBotWrap(update))
+            .flatMap(cmd -> handleExecution(cmd, update))
+            .ifPresent(this::handleSend);
     }
 
     @Override
     public String getBotUsername() {
         return this.props.botName();
     }
 
-    @Override
-    public String getBotToken() {
-        return this.props.botToken();
+    public void start() {
+        try {
+            this.api.registerBot(this);
+        } catch (final TelegramApiException exception) {
+            LOG.error("Error starting bot: {}", exception.getMessage(), exception);
+        }
     }
 
-    public void start() {
+    private static Optional<Send<AbsSender>> handleExecution(
```

Comment:
There are no reason for this method to be static

Reply:
this method not link to class state